### PR TITLE
txn: add add/drop column mode

### PR DESCRIPTION
**Pull request by cfzjywxk**
Support add/drop column modes for random tests.

### CODE REVIEW

**Comment by cfzjywxk** [comment]
@zyguan @you06 Should we move this repo into PingCAP/QE?

**Comment by you06** [comment]
> @zyguan @you06 Should we move this repo into PingCAP/QE?

Ok

File: ddl.go [hunk 1]
```diff
@@ -222,3 +223,47 @@ func dropUniqueIndex() (string, string) {
 
 	return indexName, b.String()
 }
+
+func AddColumn(columns *[]ColumnType, db *sql.DB, log *Log, readyDMLWg, readyDDLWg, readyCommitWg *sync.WaitGroup) {
+	readyDMLWg.Done()
+	threadName := "add-column"
+	util.AssertNil(log.NewThread(threadName))
+	readyDDLWg.Wait()
+	for i := 0; i < ddlCnt/2; i++ {
```
Comment (you06) [inline ddl.go:232]: Use `ddlCnt` is fine.

File: ddl.go [hunk 2]
```diff
@@ -222,3 +223,47 @@ func dropUniqueIndex() (string, string) {
...
+	util.AssertNil(log.NewThread(threadName))
+	readyDDLWg.Wait()
+	colNum := len(*columns)
+	leftIndex := 1
+	if colCnt/2 > leftIndex {
+		leftIndex = colCnt / 2
+	}
+	for i := colNum - 1; i >= leftIndex; i-- {
+		dropColumn := (*columns)[i]
+		stmt := fmt.Sprintf("ALTER TABLE %s DROP COLUMN %s", tableName, dropColumn.name)
```
Comment (you06) [inline ddl.go:259]: Check if this column is used by index is required.

**Comment by you06** [comment]
We shall refine it after the repository transfer.

**Pull request merged**